### PR TITLE
Fixed error when trying to use models with Dropout or Weight Regulari…

### PR DESCRIPTION
**Pull request by ivdorelian**
…zers

As per https://github.com/fchollet/keras/issues/2417 the train mode is explicitly passed to K.function in order to avoid errors.

### CODE REVIEW

**Comment by taoyizhi68** [comment]
still got error when I tried the snake example.
